### PR TITLE
deps: update cryptography dependency

### PR DESCRIPTION
**Pull request by lengau**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `make lint`?
- [x] Have you successfully run `pytest tests/unit`?

-----

This depends on #4131 - I'll use comments to annotate what's actually part of this and what's not.

CRAFT-1720

### CODE REVIEW

File: requirements-devel.txt
Comment (lengau) [inline requirements-devel.txt:1]: These changes are relevant here.

File: requirements.txt
Comment (lengau) [inline requirements.txt:1]: Changes relevant

File: setup.cfg
Comment (lengau) [inline setup.cfg:1]: IRRELEVANT: Changes inherited from #4131

File: setup.py
Comment (lengau) [inline setup.py:1]: IRRELEVANT: Changes inherited from #4131

File: snap/snapcraft.yaml
Comment (lengau) [inline snap/snapcraft.yaml:1]: These changes are actually part of this :-)

File: tools/freeze-requirements.sh
Comment (lengau) [inline tools/freeze-requirements.sh:1]: Yep, look at this one please :-)

File: tools/freeze-requirements.sh [hunk 1]
```diff
@@ -7,23 +7,17 @@ requirements_fixups() {
...
+  # Keep setuptools < 66
   sed -i '/^setuptools/d' "$req_file"
-  echo 'setuptools==49.6.0' >> "$req_file"
+  echo 'setuptools<66' >> "$req_file"
 
   # Pinned pyinstaller for windows.
   if [[ "$req_file" == "requirements-devel.txt" ]]; then
       sed -i '/pyinstaller/d' "$req_file"
-      echo 'pyinstaller==4.3; sys.platform == "win32"' >> "$req_file"
+      echo 'pyinstaller==4.10; sys.platform == "win32"' >> "$req_file"
```
Comment (lengau) [inline tools/freeze-requirements.sh:20]: We updated pyinstaller elsewhere but didn't change `freeze-requirements.sh` for it. I'm happy to move this one-liner to its own PR if preferred

File: tox.ini
Comment (lengau) [inline tox.ini:1]: IRRELEVANT: Changes inherited from #4131

File: tools/freeze-requirements.sh (hunk 1 above)
Comment (lengau) [reply tools/freeze-requirements.sh:20]: I did it separately: https://github.com/snapcore/snapcraft/pull/4147

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -38,6 +38,13 @@ build-packages:
   - sed
 
 parts:
+  cryptography-deps:
```
Comment (sergiusens) [inline snap/snapcraft.yaml:41]: this part is not strictly required, it might as well be part of a build-packages entry for where this is needed.

**Review by sergiusens — APPROVED** [review]

**Pull request merged**